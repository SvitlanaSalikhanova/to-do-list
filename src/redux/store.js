import { configureStore } from '@reduxjs/toolkit';
import tasksReducer from './TasksSlice';
import filterReducer from './FilterSlice';

const store = configureStore({
    reducer: {
        tasks: tasksReducer,
        filter: filterReducer,
    },
});

export default store;
