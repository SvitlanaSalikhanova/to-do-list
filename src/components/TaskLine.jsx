import React from 'react';
import { useDispatch } from 'react-redux';
import Check from './Check';
import DeleteButton from './DeleteButton';
import TaskLineCss from '../styles/TaskLine.module.scss';
import { completeTask, deleteTask } from '../redux/TasksSlice';

export default function TaskLine({ item, style }) {
    const dispatch = useDispatch();
    const onCheckChange = () => dispatch(completeTask(item.id));
    const onDelete = () => dispatch(deleteTask(item.id));
    return (
        <div
            className={TaskLineCss.taskLine}
            style={style}
        >
            <Check item={item} onChange={onCheckChange} />

            <div
                className={
                    item.isDone
                        ? TaskLineCss.completedTaskText
                        : TaskLineCss.uncompletedTaskText
                }
            >
                {item.text}
            </div>
            <div className={TaskLineCss.deleteBlock}>
                <DeleteButton onClick={onDelete} />
            </div>
        </div>
    );
}
