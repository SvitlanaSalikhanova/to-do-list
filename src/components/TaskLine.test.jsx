import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import TaskLine from './TaskLine';

const UncompletedItem = {
    id: 1,
    text: 'text',
    isDone: false,
};

const CompletedItem = {
    id: 1,
    text: 'text',
    isDone: true,
};

describe('TaskLine component tests', () => {
    it('TaskLine for Completed task Matches Snapshot', () => {
        const domTree = renderer.create(
            <Provider store={store}>
                <TaskLine item={UncompletedItem} />
            </Provider>,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('TaskLine for Uncompleted task Matches Snapshot', () => {
        const domTree = renderer.create(
            <Provider store={store}>
                <TaskLine item={CompletedItem} />
            </Provider>,
        ).toJSON();
        expect(domTree).toMatchSnapshot();
    });

    it('TaskLine checkbox for completed task is checked', () => {
        render(<Provider store={store}><TaskLine item={CompletedItem} /></Provider>);
        const check = screen.getByRole('checkbox');
        expect(check).toBeInTheDocument();
        expect(check.checked).toBe(true);
    });

    it('TaskLine checkbox for uncompleted task is unchecked', () => {
        render(<Provider store={store}><TaskLine item={UncompletedItem} /></Provider>);
        const check = screen.getByRole('checkbox');
        expect(check).toBeInTheDocument();
        expect(check.checked).toBe(false);
    });

    it('TaskLine description renders', () => {
        render(<Provider store={store}><TaskLine item={UncompletedItem} /></Provider>);
        const description = screen.getByText('text');
        expect(description).toBeInTheDocument();
    });

    it('TaskLine deleteButton renders', () => {
        render(<Provider store={store}><TaskLine item={UncompletedItem} /></Provider>);
        const button = screen.getByRole('button');
        expect(button).toBeInTheDocument();
    });
});
