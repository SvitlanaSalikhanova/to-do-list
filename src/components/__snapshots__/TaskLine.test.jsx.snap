// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskLine component tests TaskLine for Completed task Matches Snapshot 1`] = `
<div
  className="taskLine"
>
  <div
    className="container"
  >
    <div
      className="round"
    >
      <input
        checked={false}
        id="checkbox1"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="checkbox1"
      />
    </div>
  </div>
  <div
    className="uncompletedTaskText"
  >
    text
  </div>
  <div
    className="deleteBlock"
  >
    <button
      className="deleteButton"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="BsXLg"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8 2.146 2.854Z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`TaskLine component tests TaskLine for Uncompleted task Matches Snapshot 1`] = `
<div
  className="taskLine"
>
  <div
    className="container"
  >
    <div
      className="round"
    >
      <input
        checked={true}
        id="checkbox1"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="checkbox1"
      />
    </div>
  </div>
  <div
    className="completedTaskText"
  >
    text
  </div>
  <div
    className="deleteBlock"
  >
    <button
      className="deleteButton"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="BsXLg"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M2.146 2.854a.5.5 0 1 1 .708-.708L8 7.293l5.146-5.147a.5.5 0 0 1 .708.708L8.707 8l5.147 5.146a.5.5 0 0 1-.708.708L8 8.707l-5.146 5.147a.5.5 0 0 1-.708-.708L7.293 8 2.146 2.854Z"
        />
      </svg>
    </button>
  </div>
</div>
`;
